src/Akka.Persistence.Azure/Query/Publishers/CurrentEventsByTagPublisher.cs

```diff
@@ -41,14 +41,14 @@ protected override void ReceiveInitialRequest()
             Replay();
         }
 
-        protected override void ReceiveRecoverySuccess(long highestSequenceNr)
+        protected override void ReceiveRecoverySuccess(bool completed)
         {
+            if (completed)
+                _completed = true;
+            
             Buffer.DeliverBuffer(TotalDemand);
 
-            if (highestSequenceNr < ToOffset)
-                _toOffset = highestSequenceNr;
-
-            if (Buffer.IsEmpty && CurrentOffset > ToOffset)
```

Comment:
This code is a vestigial code from SQLite implementation. There is no way for the publisher to know if it received all available tagged records by highest sequence number alone. 
Also, `_toOffset = highestSequence` only works if the database only returns a single page result (the actual cause of the bug).

Reply:
Sigh, nice catch.